docs/book/architecture.rst

```diff
@@ -22,18 +22,40 @@ Sylius resource management system lives in the **SyliusResourceBundle** and can
 Services
 ````````
 
-For every resource you have three very important services available:
+For every resource you have four very essential services available:
 
+* Factory
 * Manager
 * Repository
 * Controller
 
-Let us take the "product" resource as an example. By default, It is represented by ``Sylius\Component\Core\Model\Product`` class and implement proper ``ProductInterface``.
+Let us take the "product" resource as an example. By default, it is represented by the ``Sylius\Component\Core\Model\Product`` class and it implements a proper ``ProductInterface``.
```

Comment:
What exactly does `proper interface' mean?. Which is proper? Which is not? What is life?